src/controller/python/chip/clusters/attribute.cpp

```diff
@@ -339,10 +351,12 @@ chip::ChipError::StorageType pychip_ReadClient_ReadAttributes(void * appContext,
 
     std::unique_ptr<ReadClientCallback> callback = std::make_unique<ReadClientCallback>(appContext);
 
+    size_t m = total - n;
     va_list args;
-    va_start(args, n);
+    va_start(args, total);
 
     std::unique_ptr<AttributePathParams[]> readPaths(new AttributePathParams[n]);
+    std::unique_ptr<chip::app::DataVersionFilter[]> dataVersionFilters(new chip::app::DataVersionFilter[n]);
```

Comment:
```suggestion
    std::unique_ptr<chip::app::DataVersionFilter[]> dataVersionFilters(new chip::app::DataVersionFilter[m]);
```

The fact that this passed tests is worrisome, unless we had n > m in all those tests.  And if so, we should have more tests.  Followup fine for that part.

Reply:
good catch, would follow-up more test on this part, thanks

Reply:
https://github.com/project-chip/connectedhomeip/issues/15020